lib/pbench/server/indexing_tarballs.py

```diff
@@ -0,0 +1,415 @@
+                        if ie_len > len(tb) + 1:
+                            try:
+                                report.post_status(tstos(end), "errors", ie_filepath)
+                            except Exception:
+                                idxctx.logger.exception(
+                                    "Unexpected error issuing"
+                                    " report status with errors: {}",
+                                    ie_filepath,
+                                )
+                    finally:
+                        # Unconditionally remove the indexing errors file.
+                        try:
+                            os.remove(ie_filepath)
+                        except Exception:
+                            pass
+                    # Distinguish failure cases, so we can retry the indexing
+                    # easily if possible.  Different `linkerrdest` directories for
+                    # different failures; the rest are going to end up in
+                    # `linkerrdest` for later retry.
+                    controller_path = linksrc_dir.parent
+
+                    if tb_res == 0:
+                        idxctx.logger.info(
+                            "{}: {}/{}: success",
+                            idxctx.TS,
+                            controller_path.name,
+                            os.path.basename(tb),
+                        )
+                        # Success
+                        with indexed.open(mode="a") as fp:
+                            print(tb, file=fp)
+                        rename_tb_link(
+                            tb, Path(controller_path, self.linkdest), idxctx.logger
+                        )
+                    elif tb_res == 1:
+                        idxctx.logger.warning(
+                            "{}: index failures encountered on {}", idxctx.TS, tb
+                        )
+                        with erred.open(mode="a") as fp:
+                            print(tb, file=fp)
+                        rename_tb_link(
+                            tb,
+                            Path(controller_path, f"{self.linkerrdest}.1"),
+                            idxctx.logger,
+                        )
+                    elif tb_res in (2, 3):
+                        assert False, (
+                            f"Logic Bomb!  Unexpected tar ball handling "
+                            f"result status {tb_res:d} for tar ball {tb}"
+                        )
+                    elif tb_res >= 4 or res <= 11:
+                        # # Quietly skip these errors
+                        with skipped.open(mode="a") as fp:
+                            print(tb, file=fp)
+                        rename_tb_link(
+                            tb,
+                            Path(controller_path, f"{self.linkerrdest}.{tb_res:d}"),
+                            idxctx.logger,
+                        )
+                    else:
```

Comment:
LGTM seems to be diagnosing this as unreachable. We've got `if 0 elif 1 elif in (2,3) elif >= 4` ... so, yeah, that covers everything. We can't ever hit the `else`. So what did we really intend here, again? I really do not like the error number logic in this module; it's unclear and poorly documented.

You didn't create the problem, so maybe it doesn't need to be fixed right now; but we need to look at this again.

Reply:
yeah sure

Reply:
Well, line 326 should be `elif tb_res >= 4 and tb_res <= 11:`, then this makes sense.

Reply:
@riya-17, `and` instead of `or`.

Reply:
yes that does resolve it